sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1Test.java

```diff
@@ -690,6 +781,98 @@ public void testReadFnWithBatchesExactMultiple() throws Exception {
     readFnTest(5 * QUERY_BATCH_LIMIT);
   }
 
+  @Test
+  public void testGqlTranslatorFnWithGqlQuery() throws Exception {
+    GqlQueryTranslatorFn translatorFn = new GqlQueryTranslatorFn(V_1_OPTIONS, mockDatastoreFactory);
+    DoFnTester<ValueProvider<String>, Query> doFnTester = DoFnTester.of(translatorFn);
```

Comment:
Feels like this all would be much easier if you made a static utility function for translating a gql query to Query, and unit-tested that. Unit-testing the DoFn would then not be important.